File: python/Recursion/palindrome_recursion.py
def is_palindrome(s, start, end):
    if start >= end:
        return True
    if s[start] != s[end]:
        return False
    return is_palindrome(s, start + 1, end - 1)

s = input("Enter a string: ")
if is_palindrome(s, 0, len(s) - 1):
    print("The string is a palindrome.")
else:
    print("The string is NOT a palindrome.")

File: java/Recursion/palindrome_recursion.java
import java.util.Scanner;

public class PalindromeRecursion {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter a string: ");
        String str = sc.nextLine();
        if (isPalindrome(str, 0, str.length() - 1)) {
            System.out.println("The string is a palindrome.");
        } else {
            System.out.println("The string is NOT a palindrome.");
        }
    }

    public static boolean isPalindrome(String str, int start, int end) {
        if (start >= end) return true;
        if (str.charAt(start) != str.charAt(end)) return false;
        return isPalindrome(str, start + 1, end - 1);
    }
}
